text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/MiniMessageParser.java

```diff
@@ -116,7 +117,7 @@ private void processTokens(final @NotNull StringBuilder sb, final @NotNull Strin
     }
   }
 
-  @NotNull Component parseFormat(final @NotNull String richMessage, final @NotNull ContextImpl context) {
+  @NotNull Node parseToTree(final @NotNull String richMessage, final @NotNull ContextImpl context) {
```

Comment:
This can just return an `ElementNode` -- it's not public API, and that removes the cast below.

Reply:
Sorry, forgot how classes and interfaces work